src/anyp/Uri.cc

```diff
@@ -81,6 +81,66 @@ AnyP::Uri::Encode(const SBuf &buf, const CharacterSet &ignore)
     return output;
 }
 
+SBuf
+AnyP::Uri::Rfc3986Encode(const SBuf &buf)
```

Comment:
Please just have the callers pass the CharacterSet they want encoded. Each caller case has a different set even within URI sections.

Reply:
This is still possible using the Encode function, that's not going away under this proposal.
I special-cased RFC3986 for convenience; do you still think that this use case is not common enough to warrant it?

Reply:
I think it will be not common enough, and touching the encoder should be out of scope for this PR. If I am wrong about the utility it and/or other wrappers can be easily added in a followup PR.

Reply:
> Amos: Please just have the callers pass the CharacterSet they want encoded. ... I think it will be not common enough, and touching the encoder should be out of scope for this PR. If I am wrong about the utility it and/or other wrappers can be easily added in a followup PR.

I agree.

> Francesco: Done.

I did not find any signs that the changes requested by Amos were actually implemented. Now done (branch commit 65eeca9).